sdk/core/azure-core-amqp/src/test/java/com/azure/core/amqp/implementation/handler/ReceiverUnsettledDeliveriesTest.java

```diff
@@ -81,15 +81,25 @@ public void tracksOnDelivery() throws IOException {
     }
 
     @Test
-    public void sendDispositionErrorsForUntrackedDelivery() {
+    public void sendDispositionEmitsDeliveryNotOnLinkExceptionForUntrackedDelivery() {
         try (ReceiverUnsettledDeliveries deliveries = createUnsettledDeliveries()) {
             final UUID deliveryTag = UUID.randomUUID();
             final Mono<Void> dispositionMono = deliveries.sendDisposition(deliveryTag.toString(), Accepted.getInstance());
             StepVerifier.create(dispositionMono)
-                .verifyError(IllegalArgumentException.class);
+                .verifyError(DeliveryNotOnLinkException.class);
         }
     }
 
+    @Test
+    public void sendDispositionEmitsDeliveryNotOnLinkExceptionIfClosed() {
+        ReceiverUnsettledDeliveries deliveries = createUnsettledDeliveries();
+        deliveries.close();
+        final UUID deliveryTag = UUID.randomUUID();
+        final Mono<Void> dispositionMono = deliveries.sendDisposition(deliveryTag.toString(), Accepted.getInstance());
+        StepVerifier.create(dispositionMono)
+            .verifyError(DeliveryNotOnLinkException.class);
+    }
+
```

Comment:
Line 257: I was curious why we are using an array to hole dispositionCallCounts if it is only one element? Instead of keeping track of invocations that way, we can also use:  `verify(delivery).disposition(any());`

Reply:
Inside `sendDisposition(,)` implementation, it calls `delivery.disposition(state)`, but before the mock (doAnswer hook) responds to that disposition call, it is required that `sendDisposition(,)` insert the deliveryTag into ReceiverUnsettledDeliveries::pendingDispositions Map. The `ReceiverUnsettledDeliveries::onDispositionAck` will look up the Map for the deliveryTag. So, if mock responds to disposition call with `onDispositionAck` before updating the Map, the test won’t be able to validate the scenario. This is why the mock is not responding when it is invoked upon first disposition(state) call, instead the test let the `sendDisposition` "complete it execution" then invoke `onDispositionAck(,) `below for the first disposition(state) call.

I'll add a comment to the test and slightly change the test to make it more clear.